composer.json

```diff
@@ -180,6 +182,9 @@
             "drupal/facets": {
                 "Facet Summary for Facet Items with a set dependency. @see https://www.drupal.org/node/2873480": "resources/patch/facet_summary_for_facet-2873480-18.patch"
             },
+            "drupal/filecache": {
+                "Provide an option to persist the cache. @see https://github.com/pfrenssen/filecache/pull/2": "https://patch-diff.githubusercontent.com/raw/pfrenssen/filecache/pull/2.patch"
```

Comment:
This concept of persisting cache over an cache clear is at least odd. I've never seen something similar. Cache is something that can and must be wiped out. Cache that survives its own bin "clear cache" command it's so non-API, IMHO. For me looks like an over-engineering but the problem is at the conceptual level: I've never seen a cache backend that, after clearing, the cache, is still persisting the cache items. I understand the reason why this has been added: _We need the cache items to serve as HTTP request replacements on tests_. But, that could have been accomplished with a very simple developer task:

1. Make the real request;
1. Manually copy/move the fixture files from the cache directory backend to the Behat (or PHPUnit) fixture  directory;
1. Commit them in the Git repo.

Is there something else that would require this level of complexity? And, again, not the complexity that scares me but I'm seriously questioning the concept of cache items being persisted over a cache clear.

Reply:
This cache persisting functionality also existed in the D7 version of the module, and there it was even the default strategy. I think use cases similar to ours (providing a file based cache of data provided by third parties) is one of the main use cases for even using this module.

See here for the D7 implementation of this persisting cache: https://cgit.drupalcode.org/filecache/tree/filecache.inc#n317

Reply:
That is wrong. At least, we can live with that in D7, but in D8 we have honour the cache backend contract. And the contract says:

```php
  /**
   * Deletes all cache items in a bin.
   */
  public function deleteAll();
```

Period.

But with file cache module, with this setting, when we call this method we end up with a non-empty bin. Now imagine an arbitrary module that uses this module and he needs this settings ('permanent') for given functionality. He will enable the setting in settings.php, but in other place of his application will use the same cache backend expecting normal results on a cache clear, specifically an empty bin. It's a confusing behavior. 

I see in D7 they also have there a 3rd option: persisting an amount of time (UNIX timestamp). In D8 this would overlap the `max-age` concept, which would create more confusion.

However, if you insist, I can merge this PR as it is but I cannot endorse the file cache upstream patch, which conceptually for me is not to be accepted. So, this can go with the patched module and, at some point, if anyone else will RTBC that issue, we can remove the patch from composer.

Reply:
OK I understand your point and agree. I still think this is a valuable feature for this particular module, in fact this was the reason I chose it, because it offers a persisting disk cache.

However the API contract breach should be communicated clearly to the users of the module. I will add a warning to the README and will also discuss it with Ognyan (the module maintainer) to get his input.

Reply:
I have resolved this by adding the relevant information to the README. I have also contacted the module maintainer to get his blessing.